[Issue] Panic on parsing error in azure.yaml

- [x] Make sure you've installed the latest version using [instructions](https://github.com/Azure/azure-dev?tab=readme-ov-file#installupgrade-azure-developer-cli)

**Output from `azd version`**

```plain
azd version 1.23.11 (commit 8a12ac217ec685745fa195698b3c84387c922b2b)
```

**Describe the bug**

When running `azd` in directory where an `azure.yaml` file is present, if there's an error in `azure.yaml`, `azd` will panic.

```plain
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x10 pc=0x101034fec]

goroutine 1 [running]:
github.com/azure/azure-dev/cli/azd/pkg/project.Parse({0x5ce75f533680?, 0x5ce75fbdc400?}, {0x5ce75f5294a0, 0x148})
	github.com/azure/azure-dev/cli/azd/pkg/project/project.go:95 +0x66c
github.com/azure/azure-dev/cli/azd/pkg/project.Load({0x102976d58, 0x5ce75f4e3320}, {0x5ce75f533680, 0x31})
	github.com/azure/azure-dev/cli/azd/pkg/project/project.go:160 +0x150
github.com/azure/azure-dev/cli/azd/cmd.registerCommonDependencies.func17.1()
	github.com/azure/azure-dev/cli/azd/cmd/container.go:430 +0x90
github.com/azure/azure-dev/cli/azd/pkg/lazy.(*Lazy[...]).GetValue(0x1029668e0)
	github.com/azure/azure-dev/cli/azd/pkg/lazy/lazy.go:42 +0xc0
github.com/azure/azure-dev/cli/azd/cmd.registerCommonDependencies.func38.1()
	github.com/azure/azure-dev/cli/azd/cmd/container.go:848 +0x3c
github.com/azure/azure-dev/cli/azd/pkg/lazy.(*Lazy[...]).GetValue(0x1029666a0)
	github.com/azure/azure-dev/cli/azd/pkg/lazy/lazy.go:42 +0xc0
github.com/azure/azure-dev/cli/azd/cmd.registerCommonDependencies.func37(0x0?)
	github.com/azure/azure-dev/cli/azd/cmd/container.go:839 +0x24
reflect.Value.call({0x1025c3a20?, 0x10294c178?, 0x1?}, {0x10136960a, 0x4}, {0x5ce75f6ce1b0, 0x1, 0x5ce75f669940?})
	reflect/value.go:586 +0xabc
reflect.Value.Call({0x1025c3a20?, 0x10294c178?, 0x10294c178?}, {0x5ce75f6ce1b0?, 0x0?, 0x0?})
	reflect/value.go:369 +0x94
github.com/golobby/container/v3.Container.invoke(0x1f?, {0x1025c3a20, 0x10294c178})
	github.com/golobby/container/v3@v3.3.2/container.go:103 +0x98
github.com/golobby/container/v3.(*binding).make(0x5ce75f51bad0, 0x5ce75f4e3260?)
	github.com/golobby/container/v3@v3.3.2/container.go:26 +0x34
github.com/golobby/container/v3.Container.NamedResolve(0x5ce75f4e3260, {0x10250e0c0, 0x5ce75f890d00}, {0x0, 0x0})
	github.com/golobby/container/v3@v3.3.2/container.go:238 +0x178
github.com/golobby/container/v3.Container.Resolve(...)
	github.com/golobby/container/v3@v3.3.2/container.go:224
github.com/azure/azure-dev/cli/azd/pkg/ioc.(*NestedContainer).Resolve(0x4?, {0x10250e0c0?, 0x5ce75f890d00?})
	github.com/azure/azure-dev/cli/azd/pkg/ioc/container.go:218 +0x30
github.com/azure/azure-dev/cli/azd/pkg/platform.Initialize(0x5ce75f68cec0, {0x10136e448, 0x7})
	github.com/azure/azure-dev/cli/azd/pkg/platform/platform.go:33 +0x4c
github.com/azure/azure-dev/cli/azd/cmd.newRootCmd(0x0, {0x0, 0x0, 0x5ce75f4e33b0?}, 0x5ce75f68cec0, 0x1)
	github.com/azure/azure-dev/cli/azd/cmd/root.go:497 +0x3c48
github.com/azure/azure-dev/cli/azd/cmd.NewRootCmd(...)
	github.com/azure/azure-dev/cli/azd/cmd/root.go:45
github.com/azure/azure-dev/cli/azd/cmd.ExecuteWithAutoInstall({0x102976d58, 0x5ce75f4e3320}, 0x5ce75f68cec0)
	github.com/azure/azure-dev/cli/azd/cmd/auto_install.go:403 +0x128
main.main()
	github.com/azure/azure-dev/cli/azd/main.go:130 +0x71c
```

**To Reproduce**

I was using GitHub Copilot with the Microsoft Learn MCP server and Azure skills configured and using Clause Sonnet 4.6 as the model. It generated the following `azure.yaml`:

```yaml
name: standup

services:
  api:
    host: function
    language: dotnet
    project: ./api
  web:
    # placeholder for future frontend deployment

infra:
  path: infra/main.bicep
```

Removing the `web` service fixed the issue.

**Expected behavior**

I would have expected or liked to have received an invalid configuration file error instead of a panic.

**Environment**
Information on your environment:
    * Apple macOS Tahoe 26.3.1
    
**Additional context**
Add any other context about the problem here.

--- a/cli/azd/pkg/project/project.go
+++ b/cli/azd/pkg/project/project.go
@@ -54,6 +54,10 @@ func Parse(ctx context.Context, yamlContent string) (*ProjectConfig, error) {
 		)
 	}
 
+	if err := validateParsedConfig(&projectConfig); err != nil {
+		return nil, err
+	}
+
 	projectConfig.EventDispatcher = ext.NewEventDispatcher[ProjectLifecycleEventArgs]()
 
 	if projectConfig.RequiredVersions != nil && projectConfig.RequiredVersions.Azd != nil {
@@ -189,6 +193,12 @@ func Load(ctx context.Context, projectFilePath string) (*ProjectConfig, error) {
 			continue
 		}
 		projectConfig.Hooks[hookName] = externalHookList
+	}
+
+	// Validate hooks after merging infra-module definitions so that nil entries
+	// from *.hooks.yaml files are caught before they can cause nil-pointer panics.
+	if infraHookProblems := validateHooks(projectConfig.Hooks, "infra hooks"); len(infraHookProblems) > 0 {
+		return nil, &ConfigValidationError{Issues: infraHookProblems}
 	}
 
 	if projectConfig.Metadata != nil && projectConfig.Metadata.Template != "" {

--- a/cli/azd/pkg/project/project_test.go
+++ b/cli/azd/pkg/project/project_test.go
@@ -260,6 +260,13 @@ func Test_Invalid_Project_File(t *testing.T) {
 				template: test-proj-template
 			services:
 		`,
+		"NilService":  "name: test-proj\nservices:\n  web:\n    # placeholder\n",
+		"NilResource": "name: test-proj\nresources:\n  mydb:\n    # placeholder\n",
+		"NilServiceAndResource": "name: test-proj\nservices:\n  web:\n" +
+			"    # placeholder\nresources:\n  mydb:\n    # placeholder\n",
+		"NilProjectHook": "name: test-proj\nhooks:\n  preprovision:\n",
+		"NilServiceHook": "name: test-proj\nservices:\n  web:\n" +
+			"    language: python\n    host: appservice\n    hooks:\n      predeploy:\n",
 	}
 
 	for name, test := range tests {

--- a/cli/azd/pkg/project/validate.go
+++ b/cli/azd/pkg/project/validate.go
@@ -1,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package project
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+)
+
+// ConfigValidationError is returned when the azure.yaml configuration contains
+// structural problems such as nil service, resource, or hook definitions.
+// Callers can use [errors.As] to programmatically inspect the individual Issues.
+type ConfigValidationError struct {
+	Issues []string
+}
+
+func (e *ConfigValidationError) Error() string {
+	return fmt.Sprintf("azure.yaml contains invalid configuration:\n  - %s", strings.Join(e.Issues, "\n  - "))
+}
+
+// validateParsedConfig checks a freshly-parsed ProjectConfig for nil service, resource, and
+// hook definitions that would cause nil pointer dereference panics during subsequent processing.
+// YAML sections like "web:" with only a comment or whitespace unmarshal as nil map entries,
+// which would cause nil pointer dereference panics during subsequent processing.
+//
+// The function validates Services, Resources, and Hooks at both project and service levels.
+// All problems are collected and returned in a single error so the user can fix them at once.
+func validateParsedConfig(config *ProjectConfig) error {
+	var problems []string
+
+	for key, svc := range config.Services {
+		if svc == nil {
+			problems = append(problems, fmt.Sprintf(
+				"service '%s' has an empty definition;"+
+					" expected properties such as host, language, and project",
+				key,
+			))
+			continue
+		}
+
+		problems = append(problems, validateHooks(svc.Hooks, "service '"+key+"'")...)
+	}
+
+	for key, res := range config.Resources {
+		if res == nil {
+			problems = append(problems,
+				fmt.Sprintf("resource '%s' has an empty definition; expected properties such as type", key))
+		}
+	}
+
+	problems = append(problems, validateHooks(config.Hooks, "")...)
+
+	if len(problems) > 0 {
+		// Sort for deterministic output regardless of map iteration order.
+		slices.Sort(problems)
+
+		return &ConfigValidationError{Issues: problems}
+	}
+
+	return nil
+}
+
+// validateHooks checks a HooksConfig for nil entries. When scope is non-empty it is
+// prepended to each problem description to identify the parent (e.g., "service 'web'").
+func validateHooks(hooks HooksConfig, scope string) []string {
+	var problems []string
+
+	prefix := ""
+	if scope != "" {
+		prefix = scope + " "
+	}
+
+	for hookName, hookList := range hooks {
+		if hookList == nil {
+			problems = append(problems, fmt.Sprintf(
+				"%shook '%s' has an empty definition;"+
+					" expected properties such as run or shell",
+				prefix, hookName,
+			))
+			continue
+		}
+
+		for i, hook := range hookList {
+			if hook == nil {
+				problems = append(problems,
+					fmt.Sprintf("%shook '%s' entry %d has an empty definition; expected properties such as run or shell",
+						prefix, hookName, i+1))
+			}
+		}
+	}
+
+	return problems
+}

--- a/cli/azd/pkg/project/validate_test.go
+++ b/cli/azd/pkg/project/validate_test.go
@@ -1,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package project
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/ext"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNilDefinitionsReportAllErrors(t *testing.T) {
+	yamlContent := "name: test-proj\nservices:\n  web:\n    # empty\nresources:\n  mydb:\n    # empty\n"
+	projectConfig, err := Parse(context.Background(), yamlContent)
+	require.Nil(t, projectConfig)
+	require.Error(t, err)
+
+	var validationErr *ConfigValidationError
+	require.ErrorAs(t, err, &validationErr)
+
+	require.Contains(t, err.Error(), "service 'web'")
+	require.Contains(t, err.Error(), "resource 'mydb'")
+}
+
+func TestValidateParsedConfig(t *testing.T) {
+	tests := []struct {
+		name     string
+		yaml     string
+		expected []string
+	}{
+		{
+			name:     "NilProjectLevelHook",
+			yaml:     "name: test-proj\nhooks:\n  preprovision:\n",
+			expected: []string{"hook 'preprovision'"},
+		},
+		{
+			name: "NilServiceLevelHook",
+			yaml: "name: test-proj\nservices:\n  web:\n    language: python\n" +
+				"    host: appservice\n    hooks:\n      predeploy:\n",
+			expected: []string{"service 'web' hook 'predeploy'"},
+		},
+		{
+			name: "NilHookEntryInArray",
+			yaml: "name: test-proj\nhooks:\n  preprovision:\n    - run: echo first\n    -\n",
+			expected: []string{
+				"hook 'preprovision' entry 2 has an empty definition",
+			},
+		},
+		{
+			name: "MixedNilDefinitions",
+			yaml: "name: test-proj\nhooks:\n  preprovision:\n" +
+				"services:\n  api:\n    # empty\nresources:\n  db:\n    # empty\n",
+			expected: []string{
+				"service 'api'",
+				"resource 'db'",
+				"hook 'preprovision'",
+			},
+		},
+		{
+			name: "MultipleNilServices",
+			yaml: "name: test-proj\nservices:\n  web:\n  api:\n  worker:\n",
+			expected: []string{
+				"service 'web'",
+				"service 'api'",
+				"service 'worker'",
+			},
+		},
+		{
+			name: "NilServiceSkipsHookCheck",
+			yaml: "name: test-proj\nservices:\n  web:\n",
+			expected: []string{
+				"service 'web' has an empty definition",
+			},
+		},
+		{
+			name: "ErrorMessagesAreActionable",
+			yaml: "name: test-proj\nservices:\n  web:\nresources:\n  db:\nhooks:\n  preprovision:\n",
+			expected: []string{
+				"expected properties such as host",
+				"expected properties such as type",
+				"expected properties such as run or shell",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			projectConfig, err := Parse(context.Background(), tt.yaml)
+			require.Nil(t, projectConfig)
+			require.Error(t, err)
+
+			var validationErr *ConfigValidationError
+			require.ErrorAs(t, err, &validationErr)
+			require.NotEmpty(t, validationErr.Issues)
+
+			require.Contains(t, err.Error(), "azure.yaml contains invalid configuration")
+			for _, expected := range tt.expected {
+				require.Contains(t, err.Error(), expected)
+			}
+		})
+	}
+}
+
+func TestValidateParsedConfigSortedOutput(t *testing.T) {
+	// Verify that errors are sorted alphabetically for deterministic output.
+	// Run multiple times to guard against map iteration order variance.
+	yaml := "name: test-proj\nservices:\n  zz-last:\n  mm-middle:\n  aa-first:\n  dd-fourth:\n  qq-fifth:\n"
+
+	for range 50 {
+		_, err := Parse(context.Background(), yaml)
+		require.Error(t, err)
+
+		var validationErr *ConfigValidationError
+		require.ErrorAs(t, err, &validationErr)
+
+		msg := err.Error()
+		aaIdx := strings.Index(msg, "aa-first")
+		ddIdx := strings.Index(msg, "dd-fourth")
+		mmIdx := strings.Index(msg, "mm-middle")
+		qqIdx := strings.Index(msg, "qq-fifth")
+		zzIdx := strings.Index(msg, "zz-last")
+
+		require.Greater(t, aaIdx, 0, "aa-first should be present")
+		require.Greater(t, ddIdx, aaIdx, "dd-fourth should come after aa-first")
+		require.Greater(t, mmIdx, ddIdx, "mm-middle should come after dd-fourth")
+		require.Greater(t, qqIdx, mmIdx, "qq-fifth should come after mm-middle")
+		require.Greater(t, zzIdx, qqIdx, "zz-last should come after qq-fifth")
+	}
+}
+
+// TestValidateHooksNilSlice directly exercises the hookList == nil branch in validateHooks
+// by constructing a ProjectConfig with a nil hook slice (as opposed to a slice containing nil entries).
+// This path is reachable when a *.hooks.yaml infra module file defines a hook name with no body.
+func TestValidateHooksNilSlice(t *testing.T) {
+	config := &ProjectConfig{
+		Name: "test-proj",
+		Hooks: map[string][]*ext.HookConfig{
+			"preprovision": nil,
+		},
+	}
+	err := validateParsedConfig(config)
+	require.Error(t, err)
+
+	var validationErr *ConfigValidationError
+	require.ErrorAs(t, err, &validationErr)
+	require.Len(t, validationErr.Issues, 1)
+	require.Contains(t, validationErr.Issues[0], "preprovision")
+	require.Contains(t, validationErr.Issues[0], "has an empty definition")
+}

